gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -308,74 +366,38 @@ private String getBackendFromRoutingGroup(HttpServletRequest request)
     private String getPreviousBackend(HttpServletRequest request)
     {
         String queryId = extractQueryIdIfPresent(request);
-
-        // Find query id and get url from cache
         if (!isNullOrEmpty(queryId)) {
             return routingManager.findBackendForQueryId(queryId);
         }
+        else if (!Strings.isNullOrEmpty(request.getRequestedSessionId())) {
+            //pin browser sessions to the same backend based on jsessionid, but load balance queries
```

Comment:
?? how is that related to load balancing here .. thats just what happens in fall through right?

Reply:
That comment was leftover from earlier development, removed